lib/src/inline_syntaxes/footnote_ref_syntax.dart

```diff
@@ -0,0 +1,69 @@
+import '../ast.dart' show Element, Node, Text;
+import '../charcode.dart';
+import '../inline_parser.dart' show InlineParser;
+import 'link_syntax.dart' show LinkContext;
+
+// Footnotes syntax share the same heading '[' with Reference Link, this makes
+// a lot of mass, if extends `LinkSyntax`, it will have to override
```

Comment:
This comment is fine for the PR, but it doesn't have to be checked in. It can be removed.